### ISSUE TITLE
Attempt to Reproduce the Results

### ISSUE THREAD

**Issue by xwjabc** [fundamentalvision/Deformable-DETR#8]
Recently I attempt to train the Deformable DETR model on a 8-GPU machine (8x TITAN RTX) following the command:
``` 
GPUS_PER_NODE=8 ./tools/run_dist_launch.sh 8 ./configs/r50_deformable_detr.sh
```

The results I got are shown as below:
```
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.435
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.625
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.472
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.256
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.467
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.575
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.346
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.578
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.620
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.394
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.665
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.808
Training time 2 days, 6:18:48
```
which has lower performance (around 1% lower) compared with the reported result: 

Method | Epochs | AP | APS | APM | APL 
-- | -- | -- | -- | -- | -- 
Deformable DETR | 50 | 44.5 | 27.1 | 47.6 | 59.6 

I wonder if the results I obtained are reasonable? As shown in https://github.com/fundamentalvision/Deformable-DETR/issues/1, it seems the pre-trained checkpoints are trained using batch size 32 (i.e. 2 nodes, 8 GPUs per node, 2 images per GPU). Is the performance gap due to the batch size? (Or the environment settings?) Thanks for your great help!

**Comment by jackroos** [comment]
Thanks for your interest. Do you use the latest code which we fix a small bug (commit id: 11169a60c33333af00a4849f1808023eba96a931)? The AP is slightly improved in the latest version. You could update the code and run an experiment again. And before the bug is fixed, the expected AP should be 43.8, so I think your result (with AP 43.5) is reasonable.

**Comment by xwjabc** [comment]
@jackroos  Thanks for your reply! I have not used the latest code. Will re-train the model with the updated code!

**Update**: I got AP 44.0 after switching to the new code. It seems the APs and APl are somewhat lower than the reported ones.
```
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.440
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.629
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.481
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.259
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.475
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.583
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.349
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.583
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.626
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.410
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.674
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.809
Training time 2 days, 5:22:27
```

**Comment by twangnh** [comment]
@jackroos have you reproduced the 44.5 result?

**Comment by cxxgtxy** [comment]
I meet with the same problem.  And I  use the updated code. The APs are lower than reported ones.

IoU metric: bbox
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.437
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.623
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.477
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.251
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.471
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.579
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.352
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.585
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.627
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.412
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.672
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.822

**Comment by IgorDroz** [comment]
@xwjabc @cxxgtxy Could you guys share your list of packages that you used in order to reproduce it? (i mean just copy the 'pip freeze' output). i am just unable to reproduce the results and i wonder if the problems is the version of my packages
